packages/merge/src/typedefs-mergers/directives.ts

```diff
@@ -2,85 +2,103 @@ import {
   ArgumentNode,
   DirectiveDefinitionNode,
   DirectiveNode,
+  Kind,
   ListValueNode,
   NameNode,
+  ValueNode,
 } from 'graphql';
-import { isSome } from '@graphql-tools/utils';
 import { Config } from './merge-typedefs.js';
 
-function directiveAlreadyExists(
-  directivesArr: ReadonlyArray<DirectiveNode>,
-  otherDirective: DirectiveNode,
-): boolean {
-  return !!directivesArr.find(directive => directive.name.value === otherDirective.name.value);
-}
-
 function isRepeatableDirective(
   directive: DirectiveNode,
   directives?: Record<string, DirectiveDefinitionNode>,
+  repeatableLinkImports?: Set<string>,
 ): boolean {
-  return !!directives?.[directive.name.value]?.repeatable;
+  return !!(
+    directives?.[directive.name.value]?.repeatable ??
+    repeatableLinkImports?.has(directive.name.value)
+  );
 }
 
 function nameAlreadyExists(name: NameNode, namesArr: ReadonlyArray<NameNode>): boolean {
   return namesArr.some(({ value }) => value === name.value);
 }
 
 function mergeArguments(a1: readonly ArgumentNode[], a2: readonly ArgumentNode[]): ArgumentNode[] {
-  const result: ArgumentNode[] = [...a2];
```

Comment:
To merge arguments correctly, even if defined within the same schema, we must not assume `a2` is all clear here. Instead, we must process the entire list of arguments to build up the result set.